datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/metadata.py

```diff
@@ -258,18 +252,12 @@ def metadata(check, check_duplicates, show_warnings):
     If `check` is specified, only the check will be validated,
     otherwise all metadata files in the repo will be.
     """
-    metric_sources = get_metric_sources()
-
-    if check:
-        if check not in metric_sources:
-            abort(f'Metadata file `{get_metadata_file(check)}` does not exist.')
```

Comment:
Do we still need to check for this, but move it into the `for check in checks` loop?

Reply:
On reading through the rest of the code I guess this isn't needed since you're passing `source=metrics`?

Reply:
Sorry, thinking through this, we probably still want this file existence check for the case of a user passing in a specific check, in case that doesn't have the file.

Reply:
The `echo_info` will do some of that work by indicating `0` checks if nothing gets identified.  Not sure if it's needed to add a whole if/else clause to check the results of `checks` and print a statement if its empty?

Reply:
Sorry, I don't mean the result of checks, I mean the logic of making sure the passed in `check` has a metadata.csv file defined. With these changes now, you get a stacktrace with `FileNotFoundError` if the check is present without a metadata instead of a pretty-printed error.